central/alert/service/service_impl.go

```diff
@@ -172,8 +173,35 @@ func (s *serviceImpl) GetAlertsCounts(ctx context.Context, request *v1.GetAlerts
 	if request == nil {
 		request = &v1.GetAlertsCountsRequest{}
 	}
+
 	request.Request = ensureAllAlertsAreFetched(request.GetRequest())
-	alerts, err := s.dataStore.ListAlerts(ctx, request.GetRequest())
+	var requestQ *v1.Query
+	if request.GetRequest().GetQuery() == "" {
+		requestQ = search.NewQueryBuilder().AddStringsHighlighted(search.Cluster, search.WildcardString).ProtoQuery()
+	} else {
+		var err error
+		q, err := search.ParseQuery(request.GetRequest().GetQuery())
+		if err != nil {
+			return nil, err
+		}
+		search.ApplyFnToAllBaseQueries(q, func(bq *v1.BaseQuery) {
+			matchFieldQuery, ok := bq.GetQuery().(*v1.BaseQuery_MatchFieldQuery)
+			if !ok {
+				return
+			}
+
+			if matchFieldQuery.MatchFieldQuery.GetField() == search.Cluster.String() {
```

Comment:
Is the endpoint limited to accepting cluster queries only? You should do the same for Category and Severity.

Reply:
Good point, yes. The UI uses only cluster and namespace but anyone can invoke that API with any query. Done.